app/models/machine.rb

```diff
@@ -1,5 +1,21 @@
 class Machine < ApplicationRecord
   validates_presence_of :location
-  
+
   belongs_to :owner
+  has_many :machine_snacks
+  has_many :snacks, through: :machine_snacks
+
+
+  def average_snack_price
+    if snacks.empty?
+      0
+    else
+      snacks.average(:price).round(2)
```

Comment:
I would consider rounding to be presentation logic, so I'd take that to the view. That being said though, rounding it to two will not necessarily make it look like a currency amount. You could either write a model method in that case for formatting, but since presentation is a view job, you can look into an action view helper called `number_to_currency`.